datasetinsights/estimators/faster_rcnn.py

```diff
@@ -22,6 +22,605 @@
+            }
+            self.writer.add_scalars(
+                f"val/{metric_name}-per-class", label_results, epoch
+            )
+            fig = metric_per_class_plot(metric_name, result, label_mappings)
+            self.writer.add_figure(f"{metric_name}-per-class", fig, epoch)
+
+    def save(self, path):
+        """ Serialize Estimator to path
+
+        Args:
+            path (str): full path to save serialized estimator
+
+        Returns:
+            saved full path of the serialized estimator
+        """
+        save_dict = {
+            "model": self.model_without_ddp.state_dict(),
+            "config": self.config,
+        }
+        torch.save(save_dict, path)
+
+        return path
+
+    def load(self, path):
+        """ Load Estimator from path
+
+        Args:
+            path (str): full path to the serialized estimator
+        """
+        logger.info(f"loading checkpoint from file")
+        checkpoint = torch.load(path, map_location=self.device)
+        self.model_without_ddp.load_state_dict(checkpoint["model"])
+        loaded_config = copy.deepcopy(checkpoint["config"])
+        stored_config = copy.deepcopy(self.config)
+        del stored_config["checkpoint_file"]
+        if stored_config != loaded_config:
+            logger.debug(
+                f"Found difference in estimator config."
+                f"Estimator loaded from {path} was trained using config: \n"
+                f"{loaded_config}. \nHowever, the current config is: \n"
+                f"{self.config}."
+            )
+        self.model_without_ddp.eval()
+
+    def predict(self, pil_img, box_score_thresh=0.5):
+        """ Get prediction from one image using loaded model
+
+        Args:
+            pil_img (PIL Image): PIL image from dataset.
+            box_score_thresh (float): box score threshold for filter out lower
+            score bounding boxes. Defaults to 0.5.
+
+        Returns:
+            filtered_pred_annotations (List[BBox2D]): high predicted score
+            bboxes from the model.
+        """
+        pil_img = pil_img.unsqueeze(0)
+        img_tensor = pil_img.to(self.device)
+        predicts = self.model_without_ddp(img_tensor)
+        predict_annotations = convert_bboxes2canonical(predicts)[0]
+        filtered_pred_annotations = [
+            box for box in predict_annotations if box.score >= box_score_thresh
+        ]
+        return filtered_pred_annotations
+
+
+def create_dryrun_dataset(config, dataset, split):
+    """
+    create dataset with very small no of images
+    Args:
+        config: (CfgNode): estimator config:
+        dataset: dataset obj must have len and __get_item__
+        split: train, val, test
+
+    Returns dataset: Subset dataset
+
+    """
+    r = np.random.default_rng()
+    indices = r.integers(0, len(dataset), config[split].batch_size * 2)
+    dataset = torch.utils.data.Subset(dataset, indices)
+    return dataset
+
+
+def download_data(config, split):
```

Comment:
Why do we need a 1-line method to download dataset? This method seems to be over-engineering.

Reply:
It seems one line, but it has alot parameters so makes bit lengthy main method and naming(Dataset.create) makes bit unreadable so fix both created separate method.

Reply:
Can the method name be improved? It's very ambiguous to see a name like "download_data" as this not what this method does.